pkg/client/unversioned/remotecommand/v1.go

```diff
@@ -28,19 +28,27 @@ import (
 )
 
 // streamProtocolV1 implements the first version of the streaming exec & attach
-// protocol. This version has some bugs, such as not being able to detecte when
+// protocol. This version has some bugs, such as not being able to detect when
 // non-interactive stdin data has ended. See http://issues.k8s.io/13394 and
 // http://issues.k8s.io/13395 for more details.
 type streamProtocolV1 struct {
-	stdin  io.Reader
-	stdout io.Writer
-	stderr io.Writer
-	tty    bool
+	StreamOptions
+
+	errorStream  httpstream.Stream
```

Comment:
I spent some time a few months ago trying to do an http2 implementation of the httpstream interfaces. I'm not sure it's going to be possible, or at least it won't be really easy. It's been a few months since I wrote this PR, so I don't remember the exact reason why I switched from io.Reader/Writer to httpstream.Stream. I'd have to go back and look.